### ISSUE TITLE
Add option on `HDF5IO` to load namespaces without warning or use warning subclass

### ISSUE THREAD

**Issue by rly** [hdmf-dev/hdmf#602]
## Description

Currently when using `HDF5IO` with keyword argument `load_namespaces=True` to read a file and its cached schema, and the cached schema version differs from the loaded schema version, HDMF raises a warning saying:

UserWarning: Ignoring cached namespace 'core' version 2.0.2 because version 2.2.5 is already loaded.

It would be nice to be able to suppress these warnings, either through a keyword argument on `HDF5IO` e.g., `warn=False` or by defining a custom Warning class for this warning, e.g., `NamespaceLoadedWarning` that can be filtered in user code using the Python warnings module.

**Comment by t-b** [comment]
Suppressing the warning sounds like a workaround or? Wouldn't it be better if there would be no global loaded schema and only the loaded one from the file being present?

**Comment by rly** [comment]
Yes this is a workaround until we implement better, more explicit loading of namespaces, which is a significant undertaking.
